waltz-ng/client/logical-flow/components/bulk-logical-flow-parser/bulk-logical-flow-parser.js

```diff
@@ -0,0 +1,201 @@
+/*
+ * Waltz - Enterprise Architecture
+ * Copyright (C) 2016, 2017  Waltz open source project
+ * See README.md for more information
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import _ from 'lodash';
+import { nest } from "d3-collection";
+import { CORE_API } from '../../../common/services/core-api-utils';
+import { initialiseData } from '../../../common';
+import { refToString, toEntityRef } from '../../../common/entity-utils';
+
+
+import template from './bulk-logical-flow-parser.html';
+import { invokeFunction } from '../../../common/index';
+
+
+const bindings = {
+    columnMappings: '<',
+    sourceData: '<',
+    onParseComplete: '<'
+};
+
+
+const initialState = {
+    columnMappings: {},
+    loading: false,
+    parsedData: [],
+    sourceData: [],
+
+    onParseComplete: (event) => console.log('default onParseComplete handler for bulk-logical-flow-parser: ', event)
+};
+
+
+function resolveEntityRef(entitiesByIdentifier = {}, kind, identifier) {
+    const search = _.toLower(identifier);
+    const app = entitiesByIdentifier[search];
+    const entityRef = app ? toEntityRef(app, kind) : null;
+
+    return {
+        identifier,
+        entityRef
+    };
+}
+
+
+function mapColumns(columnMappings = {}, sourceData = [], columnResolvers = {}) {
+    const sourceKeys = _.keys(columnMappings);
```

Comment:
nitpick:   perhaps better to refer to this var as `sourceColumns` to keep inline with the rest of the naming scheme

Reply:
this is the source data (rows + cols), not just the column headings